### PR TITLE
ADEN-3775 Move SP listeners

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: extensions/wikia/AdEngine/AdEngine2Hooks.class.php [hunk 1]
```diff
@@ -98,7 +98,12 @@ public static function onWikiaSkinTopScripts( &$vars, &$scripts ) {
 		$vars['ads'] = [
 			'context' => $adContext,
 			'runtime' => [
-				'sp' => []
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/AdEngine2Hooks.class.php:101]: Because json_encode did array (`[]`) instead of object (`{}`) 🙈

**Comment by bognix** [comment]
LGTM

File: extensions/wikia/AdEngine/js/SourcePointDetection.js [hunk 2]
```diff
@@ -62,20 +63,22 @@ define('ext.wikia.adEngine.sourcePointDetection', [
...
-			window.ads.runtime.sp.blocking = false;
-			trackStatusOnce('no');
-		});
-
 		log('Appending detection script to head', 'debug', logGroup);
 		node.parentNode.insertBefore(detectionScript, node);
 		detectionInitialized = true;
 	}
 
+	win.ads.runtime.sp = win.ads.runtime.sp || {};
```
Comment (drets) [inline extensions/wikia/AdEngine/js/SourcePointDetection.js]: What if `runtime` object doesn't exist?

File: extensions/wikia/AdEngine/js/SourcePointDetection.js [hunk 3]
```diff
@@ -62,20 +63,22 @@ define('ext.wikia.adEngine.sourcePointDetection', [
...
+
+	doc.addEventListener('sp.blocking', function () {
+		win.ads.runtime.sp.blocking = true;
+		trackStatusOnce('yes');
+	});
+	doc.addEventListener('sp.not_blocking', function () {
+		win.ads.runtime.sp.blocking = false;
+		trackStatusOnce('no');
+	});
```
Comment (drets) [inline extensions/wikia/AdEngine/js/SourcePointDetection.js]: I don't like the thing that the code is executed just during module loading.
What do you think about wrapping this code in the method and call the method outside of this module?

File: extensions/wikia/AdEngine/AdEngine2Hooks.class.php [hunk 4]
```diff
@@ -98,7 +98,12 @@ public static function onWikiaSkinTopScripts( &$vars, &$scripts ) {
...
+				'pf' => [
+					'blocking' => null
+				],
+				'sp' => [
+					'blocking' => null
```
Comment (drets) [inline extensions/wikia/AdEngine/AdEngine2Hooks.class.php]: So from now we always will have `blocking` variables inside `runtime` object?

Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/AdEngine2Hooks.class.php]: Is it an issue?

Comment (drets) [inline extensions/wikia/AdEngine/AdEngine2Hooks.class.php]: Minor one!
I believe `null` values are more error prone and in order to check detection in console I should type more chars :wink:

File: extensions/wikia/AdEngine/AdEngine2ContextService.class.php [hunk 5]
```diff
@@ -50,7 +50,7 @@ public function getContext( Title $title, $skinName ) {
 					'showAds' => $adPageTypeService->areAdsShowableOnPage(),
 					'trackSlotState' => $wg->AdDriverTrackState,
 					'sourcePointDetectionUrl' => $sourcePointDetectionUrl,
-					'sourcePointRecovery' => ARecoveryModule::isEnabled(),
+					'sourcePointRecovery' => $skinName === 'oasis' && ARecoveryModule::isEnabled(),
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/AdEngine2ContextService.class.php:53]: Worth updating docs we'll enable it only on desktop/oasis.

File: extensions/wikia/AdEngine/js/context/adContext.js [hunk 6]
```diff
@@ -86,10 +86,6 @@ define('ext.wikia.adEngine.adContext', [
 				geo.isProperGeo(instantGlobals.wgAdDriverSourcePointDetectionMobileCountries));
 		}
 
-		// SourcePoint recovery integration (set in AdEngine2ContextService based on wgEnableUsingSourcePointProxyForCSS)
-		if (isUrlParamSet('sourcepointrecovery')) {
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/context/adContext.js:90]: Does it make sense to be consistent and remove `sourcepointdetection` URL param as well? Or we want to keep both parameters and this logic needs to be moved somewhere else, for example to the [init method](https://github.com/Wikia/app/pull/11052/files#diff-986e3632346ab3e613add8f5a1024b46R55)? Please updated docs too if you decide to remove any of those paramters ;)

**Comment by drets** [comment]
LGTM :+1:

**Comment by drets** [comment]
lgtm

**Pull request merged**